Freshness Edit: dotnet content (#45192)
dotnet/docs: docs/csharp/programming-guide/delegates/index.md

---
title: Work with delegate types in C#

> description: Explore delegate types in C# and refer to methods with a parameter list and return type, or pass methods as arguments to other methods.

Reviewer: The proposed edit makes this incorrect. I suggest this instead:

```suggestion
description: Explore delegate types in C#. A delegate is a date type that refers to a method with a defined parameter list and return type. You use delegates to pass methods as arguments to other methods.
```

ms.date: 03/11/2025
helpviewer_keywords: 
  - "C# language, delegates"
  - "delegates [C#]"
---
# Delegates (C# Programming Guide)

> A [delegate](../../language-reference/builtin-types/reference-types.md) is a type that represents references to methods with a particular parameter list and return type. When you instantiate a delegate, you can associate the delegate instance with any method by using a compatible signature and return type. You can invoke (or call) the method through the delegate instance.

Reviewer: Some word choice changes:

```suggestion
A [delegate](../../language-reference/builtin-types/reference-types.md) is a type that represents references to methods with a particular parameter list and return type. When you instantiate a delegate, you can associate the delegate instance with any method that has a compatible signature and return type. You can invoke (or call) the method through the delegate instance.
```

- Methods don't have to match the delegate type exactly. For more information, see [Using Variance in Delegates](../concepts/covariance-contravariance/using-variance-in-delegates.md).
- Lambda expressions are a more concise way of writing inline code blocks. Lambda expressions (in certain contexts) are compiled to delegate types. For more information about lambda expressions, see [Lambda expressions](../../language-reference/operators/lambda-expressions.md).

## Review related articles

For more information about delegates, see the following articles:

- [Using delegates](./using-delegates.md)
- [When to use delegates instead of interfaces (C# Programming Guide)](/previous-versions/visualstudio/visual-studio-2010/ms173173(v=vs.100))
- [Delegates with named versus anonymous methods](./delegates-with-named-vs-anonymous-methods.md)
- [Using variance in delegates](../concepts/covariance-contravariance/using-variance-in-delegates.md)
- [How to combine delegates (multicast delegates)](./how-to-combine-delegates-multicast-delegates.md)
- [How to declare, instantiate, and use a delegate](./how-to-declare-instantiate-and-use-a-delegate.md)

## Access the C# language specification

> The language specification is the definitive source for C# syntax and usage. For more information, see [Delegates](~/_csharpstandard/standard/delegates.md) in the [C# Language Specification](~/_csharpstandard/standard/README.md). 

Reviewer: The extra space here is causing the lint issue:

```suggestion
The language specification is the definitive source for C# syntax and usage. For more information, see [Delegates](~/_csharpstandard/standard/delegates.md) in the [C# Language Specification](~/_csharpstandard/standard/README.md).
```
